core/config/project_settings.cpp

```diff
@@ -1361,6 +1358,18 @@ void ProjectSettings::get_argument_options(const StringName &p_function, int p_i
 }
 #endif
 
+void ProjectSettings::set_editor_setting_override(const String &p_setting, const Variant &p_value) {
+	set_setting(EDITOR_SETTING_OVERRIDE_PREFIX + p_setting, p_value);
+}
+
+bool ProjectSettings::has_editor_setting_override(const String &p_setting) const {
+	return has_setting(EDITOR_SETTING_OVERRIDE_PREFIX + p_setting);
+}
+
+Variant ProjectSettings::get_editor_setting_override(const String &p_setting) const {
+	return get_setting(EDITOR_SETTING_OVERRIDE_PREFIX + p_setting);
+}
```

Comment:
This API can be exposed to scripting. There was a request for this in the proposal comments, a plugin that needs a certain editor setting value, should be able to override the editor setting when the plugin is enabled.

Reply:
This only adds a project override for a setting, it's not really to setting or getting settings. Though we might need a method to get setting with override.